electron/channels/generate-diograph-channel.js

```diff
@@ -1,26 +1,26 @@
 const { ipcMain } = require('electron')
+const UserDataStore = require('electron-store')
 const { channels } = require('../../src/shared/constants')
 const { generateDiograph } = require('../generators/diograph-generator')
 const { saveRoom } = require('../lib/room-util')
-const UserDataStore = require('electron-store')
 
 ipcMain.on(channels.GENERATE_DIOGRAPH, (event, path) => {
   console.log('GENERATE_DIOGRAPH', path)
-  generateDiograph(path).then(({ id, diograph }) => {
-    console.log(id, diograph)
+  generateDiograph(path).then(({ roomId, diograph }) => {
+    console.log(roomId, diograph)
     console.log('--------')
     saveRoom(path, diograph)
       .then(() => {
+        // Store current roomId to user data
         const store = new UserDataStore()
         store.set({
```

Comment:
I would keep every channel simple and the most of the logic in the front-end. Thus we could have a separate channel for storing necessary information to backend.

Reply:
Is there any other logic in backend than individual channels?
* GENERATE_DIOGRAPH
* GET_DIOGRAPH / GET_ROOM
* SAVE_DIOGRAPH / SAVE_ROOM
* GET_PATH / GET_USER_DATA
* SET_PATH / SET_USER_DATA

Should we treat the backend like API? Organise everything based on channels? Channels would be kind of API endpoints 🤔 And backend is nothing else than an "channel API". Then we could also make a "API docs" for it: request parameters, response example etc.

Other endpoints would get data from folders, others from user data... But for frontend it wouldn't make any difference.

Reply:
This won't be done by now. Let's consider and change the architecture later...

Also the naming will be reconsidered later.